### PR TITLE
Legal Screen for Debug Log (EXPOSUREAPP-5029)

### PR DESCRIPTION
**Pull request by axelherbstreith**
Added legal Screen for the Debug Log Feature

### CODE REVIEW

**Review by janetback — APPROVED** [review]
UA reviewed

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/legal/DebugLogLegalFragment.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        binding.apply {
+            toolbar.setOnClickListener {
+                vm.onBackButtonPress()
+            }
+        }
+
+        vm.routeToScreen.observe2(this) {
+            when (it) {
+                DebugLogLegalNavigationEvents.NavigateToShareFragment -> {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/legal/DebugLogLegalFragment.kt:35]: For simple navigation, do we need the extra `*Events` class?

File: Corona-Warn-App/src/main/res/layout/bugreporting_legal_fragment.xml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (d4rken) [inline Corona-Warn-App/src/main/res/layout/bugreporting_legal_fragment.xml:2]: Do we need the `<layout`?

File: Corona-Warn-App/src/main/res/layout/include_debuglog_legal_privacy_card.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (d4rken) [inline Corona-Warn-App/src/main/res/layout/include_debuglog_legal_privacy_card.xml:2]: Do we need the `<layout`?

File: Corona-Warn-App/src/main/res/values-de/strings.xml [hunk 4]
```diff
@@ -901,6 +901,19 @@
     <!-- XHED: Title for native sharing dialog -->
     <string name="debugging_debuglog_sharing_dialog_title">"CWA Fehlerbericht teilen"</string>
 
+    <!-- XHED: Title for debug legal screen -->
+    <string name="debugging_debuglog_legal_dialog_title">"Ausführliche Informationen zur Übersendung der Fehlerberichte"</string>
```
Comment (d4rken) [inline Corona-Warn-App/src/main/res/values-de/strings.xml:905]: @harambasicluka  @janetback  Do these need to go into the `legal_strings.xml`?

**Review by d4rken — CHANGES_REQUESTED** [review]

File: Corona-Warn-App/src/main/res/values-de/strings.xml (hunk 4 above)
Comment (AlexanderAlferov) [reply Corona-Warn-App/src/main/res/values-de/strings.xml:905]: Previously I have to add title to `legal_strings.xml` too.

**Comment by AlexanderAlferov** [comment]
Tested on Samsung S8. Aside of Matthias comments - looks good!

File: Corona-Warn-App/src/main/res/layout/bugreporting_legal_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/content_container"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@drawable/contact_diary_onboarding_background"
+    android:contentDescription="@string/contact_diary_title"
+    android:focusable="true">
+
+    <Toolbar
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/bugreporting_legal_fragment.xml:12]: Please use `MaterialToolbar`?

File: Corona-Warn-App/src/main/res/layout/bugreporting_legal_fragment.xml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    <ScrollView
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:clipToPadding="false"
+        android:fillViewport="true"
+        android:paddingBottom="8dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/toolbar">
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/bugreporting_legal_fragment.xml:32]: `@id/`

File: Corona-Warn-App/src/main/res/layout/include_debuglog_legal_privacy_card.xml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        style="@style/subtitle"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/spacing_normal"
+        android:accessibilityHeading="true"
+        android:focusable="true"
+        android:text="@string/debugging_debuglog_legal_privacy_card_second_section"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/debug_log_legal_privacy_card_first_section" />
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/include_debuglog_legal_privacy_card.xml:47]: @+id/

File: Corona-Warn-App/src/main/res/layout/include_debuglog_legal_privacy_card.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        style="@style/subtitle"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/spacing_normal"
+        android:accessibilityHeading="true"
+        android:focusable="true"
+        android:text="@string/debugging_debuglog_legal_privacy_card_first_section"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/debug_log_legal_privacy_card_title" />
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/include_debuglog_legal_privacy_card.xml:34]: @+id/

File: Corona-Warn-App/src/main/res/values-de/strings.xml (hunk 4 above)
Comment (axelherbstreith) [reply Corona-Warn-App/src/main/res/values-de/strings.xml:906]: is changed

**Review by d4rken — APPROVED** [review]

**Pull request merged**